extensions/interactions/MultipleChoiceInput/protractor.js

```diff
@@ -19,16 +19,51 @@
 
 var forms = require(process.cwd() + '/core/tests/protractor_utils/forms.js');
 
+
 // The members of richTextInstructionsArray are functions, one for each option,
 // which will each be passed a 'handler' that they can use to edit the
 // rich-text area of the option, for example by
 //   handler.appendUnderlineText('emphasised');
-var customizeInteraction = function(elem, richTextInstructionsArray) {
-  forms.ListEditor(elem).setLength(richTextInstructionsArray.length);
-  for (var i = 0; i < richTextInstructionsArray.length; i++) {
-    var richTextEditor = forms.ListEditor(elem).editItem(i, 'RichText');
-    richTextEditor.clear();
-    richTextInstructionsArray[i](richTextEditor);
+// The second arguement(optional) editInstructions is used to edit the
+// interaction after its creation. The editInstructions contains two elements,
+// the first is a command('edit', 'delete', 'add') and second element is the
+// index of the object to be edited or deleted. In the case of 'add',
+// the editInstructions has only one element.
+var customizeInteraction = function(
+    elem, richTextInstructionsArray, editInstructions) {
+  if (editInstructions === undefined) {
+    forms.ListEditor(elem).setLength(richTextInstructionsArray.length);
+    for (var i = 0; i < richTextInstructionsArray.length; i++) {
+      var richTextEditor = forms.ListEditor(elem).editItem(i, 'RichText');
+      richTextEditor.clear();
+      richTextInstructionsArray[i](richTextEditor);
+    }
+  } else {
+    if (
+      editInstructions[0] === 'delete' &&
+         richTextInstructionsArray === null) {
+      forms.ListEditor(elem).deleteItem(editInstructions[1]);
+    } else if (
+      editInstructions[0] === 'add' &&
+         richTextInstructionsArray !== null &&
+         richTextInstructionsArray.length === 1) {
```

Comment:
What if someone mis-uses this function and `editInstructions[0]` is `'add'`, but `richTextInstructionsArray` is null or has multiple elements in it? We should throw an error instead of just making no changes, right?

Reply:
done